lib/reline/config.rb

```diff
@@ -232,7 +238,9 @@ def handle_directive(directive, file, no)
     when 'if'
       condition = false
       case args
-      when 'mode'
+      when /^mode=(vi|emacs)$/i
+        condition = true
+        @editing_mode_section = $1.downcase == 'emacs' ? 'emacs' : 'vi'
```

Comment:
`condition` should be true only when editing_mode_label matches arg.

```
"\e\C-j": vi-editing-mode
"\C-e": emacs-editing-mode

set editing-mode vi
set keymap emacs
$if mode=emacs
  # skipped because editing-mode(==vi) is not emacs
  "x": "A"
$else
  # executed. set emacs-keymap "x" to "B" (insert "B")
  "x": "B"
$endif
# type ESC C-e to switch to emacs mode, type x, then B is inserted
```

```
"\e\C-j": vi-editing-mode
"\C-e": emacs-editing-mode

set editing-mode emacs
set keymap vi
$if mode=vi
  # skipped because editing-mode(==emacs) is not vi
  "x": "A"
$else
  # executed. set vi-keymap "x" to "iB" (switch to insert mode and insert "B")
  "x": "iB"
$endif
# type ESC C-j to switch to vi-insert, type ESC to switch to vi-command, type x, then B is inserted
```

And looks like key is registered to keymap (`== @keymap_label)`, so I think we don't need to introduce `@editing_mode_section`.

Reply:
Indeed, you're right. I've fixed it!